SimCalorimetry/HGCalSimProducers/interface/HGCDigitizerBase.h

```diff
@@ -131,8 +137,13 @@ class HGCDigitizerBase {
   //front-end electronics model
   std::unique_ptr<HGCFEElectronics<DFr> > myFEelectronics_;
 
+  static const int samplesize_ = hgc::nSamples;
+  const double NoiseMean_, NoiseStd_;
+  static const long NoiseArrayLength_ = 50000;
```

Comment:
This should be at least doubled, if not more, to maintain statistical independence of hits in pileup mixing.

A more robust strategy would be to roll a new set of random numbers in every layer.

Reply:
Or even to pre-cache every layer, and then mix the layers per-event. The vectorization alone will be a win, despite the caching. 

The major thing I'm worried about is the fact that you're repeating the same 50k numbers in a calorimeter that is largely quiet for most of its solid angle. Especially since this strategy is prone to underestimating noise in such cases (exponential distribution...).